### PR TITLE
Remove ServiceBroker usage from FileExtensionProvider

### PR DESCRIPTION
**Pull request by garbear**
This supersedes #13470. It contains the fix posted lower in the thread.

## How Has This Been Tested?
Compiles successfully. Unit tests don't segfault.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: xbmc/ServiceManager.cpp [hunk 1]
```diff
@@ -72,24 +72,33 @@ bool CServiceManager::InitForTesting()
...
-
+  m_binaryAddonManager.reset(new ADDON::CBinaryAddonManager());
   m_addonMgr.reset(new ADDON::CAddonMgr());
   if (!m_addonMgr->Init())
   {
     CLog::Log(LOGFATAL, "CServiceManager::%s: Unable to start CAddonMgr", __FUNCTION__);
     return false;
   }
 
+  if (!m_binaryAddonManager->Init())
```
Comment (notspiff) [inline xbmc/ServiceManager.cpp]: this needs databases afaict ? through the call to GetInstalledBinaryAddons -> CAddonMgr::GetInstalledAddons -> CAddonMgr::GetAddonsInternal -> AddonManager.cpp:587

Comment (garbear) [reply xbmc/ServiceManager.cpp]: It needs add-on DB, which is created/updated when `m_databaseManager.reset(new CDatabaseManager);` is called above

Comment (notspiff) [reply xbmc/ServiceManager.cpp]: alright.

Comment (garbear) [reply xbmc/ServiceManager.cpp]: This occurs in the constructor of DB manager. I considered breaking it out into an explicit InitializeAddonDB() function, but figured the extra code wasn't worth saving you a github comment ;)

Comment (garbear) [reply xbmc/ServiceManager.cpp]: > to me it looked like you moved this to level 1 and db's were still at level 2.

`InitForTesting()` eschews levels for some reason

Comment (notspiff) [reply xbmc/ServiceManager.cpp]: there's good reasons for that but that was indeed what confused me.

**Comment by notspiff** [comment]
does the job it's supposed to do.

**Pull request merged**